src/sage/modules/with_basis/representation.py

```diff
@@ -750,12 +750,27 @@ def __classcall_private__(cls, reps, **options):
             True
             sage: tensor([L, tensor([S, R])]) == tensor([L, S, R])
             True
+
+        Check that the tensor product with more general modules
+        can be constructed::
+
+            sage: C = CombinatorialFreeModule(ZZ, ['a','b'])
+            sage: T = tensor([C, R])
+            sage: type(T)
+            <class 'sage.combinat.free_module.CombinatorialFreeModule_Tensor_with_category'>
+            sage: T = tensor([R, C])
+            sage: type(T)
+            <class 'sage.combinat.free_module.CombinatorialFreeModule_Tensor_with_category'>
         """
-        assert (len(reps) > 0)
-        R = reps[0].base_ring()
+        assert len(reps) > 0
+        assert isinstance(reps[0], Representation_abstract)
```

Comment:
Is there a reason to an `assert` here instead of raising a `ValueError`?

Reply:
Yes, this should never occur. This class is constructed from `tensor([A,B,C])` -> `A.tensor(B,C)`, so there must be at least one such object. It's possible a user directly calls this, but no matter what, some information about what this is a representation for must be specified, which has to be given through the input modules.